src/adapter/src/coord/peek.rs

```diff
@@ -202,12 +203,46 @@ pub fn create_fast_path_plan<T: timely::progress::Timestamp>(
                 mir.typ(),
             )));
         } else {
+            if let MirRelationExpr::Project { input, outputs } = mir {
+                if outputs.iter().copied().eq(0..input.arity()) {
+                    // Trivial projection, jump through it.
+                    mir = input;
+                }
+            }
+            // If there is a TopK that would be completely covered by the finishing, then jump
```

Comment:
Can this be in an optimizer step?

Reply:
> Can this be in an optimizer step?

The problem is that the optimizer doesn't see the finishing at the moment. It could be wired up, but it would be a slightly bigger refactoring. I thought having it only here in `create_fast_path_plan` is an ok temporary solution, as the most important point of https://github.com/MaterializeInc/materialize/issues/8194 is to not lose fast path when there is an extra TopK.

Reply:
Also note that if there are multiple redundant TopKs for some reason, then TopK fusion will fuse them.